src/psyclone/psyir/symbols/symbol.py

```diff
@@ -89,20 +212,35 @@ class Visibility(Enum):
     # given an explicit visibility.
     DEFAULT_VISIBILITY = Visibility.PUBLIC
 
-    def __init__(self, name, visibility=DEFAULT_VISIBILITY):
+    def __init__(self, name, visibility=DEFAULT_VISIBILITY, interface=None):
 
         if not isinstance(name, six.string_types):
             raise TypeError(
                 "{0} 'name' attribute should be of type 'str'"
-                " but '{1}' found.".format(type(self).__name__, type(name)))
+                " but '{1}' found.".format(
+                    type(self).__name__, type(name).__name__))
         if not isinstance(visibility, Symbol.Visibility):
             raise TypeError(
                 "{0} 'visibility' attribute should be of type "
                 "psyir.symbols.Symbol.Visibility but '{1}' found.".format(
                     type(self).__name__, type(visibility).__name__))
+        if interface and not isinstance(interface, SymbolInterface):
```

Comment:
This is already checked in the interface setter, maybe we don't need this condition here?

Reply:
Removed and call the setter from init.